extensions/git/src/commands.ts

```diff
@@ -1704,9 +1720,10 @@ export class CommandCenter {
 					throw err;
 				}
 
-				const force = localize('force', "Force Checkout");
-				const stash = localize('stashcheckout', "Stash & Checkout");
-				const choice = await window.showWarningMessage(localize('local changes', "Your local changes would be overwritten by checkout."), { modal: true }, force, stash);
+				const message = opts?.command === 'switch' ? localize('local changes switch', "Your local changes would be overwritten by switch.") : localize('local changes checkout', "Your local changes would be overwritten by checkout.");
+				const force = opts?.command === 'switch' ? localize('forceswitch', "Force Switch") : localize('forcecheckout', "Force Checkout");
+				const stash = opts?.command === 'switch' ? localize('stashswitch', "Stash & Switch") : localize('stashcheckout', "Stash & Checkout");
```

Comment:
nit-pick: The difference use of single and double quotes on the above to line (BTW: I am not sure on what the project coding standard / linting enforces here)